### DISCUSSION TITLE
How to modify the links origin?

### DISCUSSION THREAD

**Discussion by eirtech** [stack-of-tasks/pinocchio#2032] [Q&A] [answered]
Hello,

I wanted to know if there is the possibility to update the origin of each link after loading the URDF? (e.g. visual/collision origin)

Thanks in advance.

**Comment by jcarpent** [comment, marked as answer]
You can do that by updating the placement values contained in `model.jointPlacements[joint_id]`, where `joint_id` is the index of the joint.

**Reply by eirtech** [reply]
Yes, I did it for example for the further FK calculation or so. But I want to change the visualization/collision properties in the URDF, too. For example, consider that I have a cylinder between the joint A and B. And joint A is in `(0, 0, 0)` and joint B is in `(0.4, 0, 0)`. So, the link AB is: 

`
<link name="AB">

        <inertial>

            <mass value="2"/>

            <origin xyz="0.2 0 0" rpy="1.57 0 1.57" />

            <inertia ixx="0.01" iyy="0.0008" izz="0.01" ixy="0" ixz="0" iyz="0"/>

        </inertial>

        <visual>

            <origin xyz="0.2 0 0" rpy="1.57 0 1.57" />

            <geometry>

                <cylinder length="0.4" radius="0.05"/>

            </geometry>

            <material name="grey">

                <color rgba="0.0 0.0 0.0 1.0"/>

            </material>

        </visual>

        <collision>

            <origin xyz="0.2 0 0" rpy="1.57 0 1.57" />

            <geometry>

                <cylinder length="0.4" radius="0.05"/>

            </geometry>

            <material name="grey">

                <color rgba="0.0 0.0 0.0 1.0"/>

            </material>

        </collision>

    </link>

`


So then by updating the placement value the joint is updated and it is good for the further calculation. 
      `model.jointPlacements[model.getJointId("B")].translation = np.array([0,2, 0, 0])`

But, the cylinder visualization and other properties, like its center that should be updated to (0.1, 0, 0) and the length that should be updated to 0.2, won't change. 

So, my question is if there is a way to update them, too?

**Reply by jcarpent** [reply]
You will have to do it in the collision and visual models associated to the joint in question.

**Reply by eirtech** [reply]
Oh thanks, I got it.
